reporter/src/main/kotlin/model/EvaluatedModel.kt

```diff
@@ -19,7 +19,284 @@
+ * * When modifying the model make sure that the objects are serialized at the right place. By default Jackson
+ *   serializes an Object with [ObjectIdInfo] the first time the serializer sees the object. If this is not desired
+ *   because the object shall be serialized as the generated ID, the [JsonIdentityReference] annotation can be used to
+ *   enforce this. For example, the list of [EvaluatedOrtIssue]s is serialized before the list of [EvaluatedPackage]s.
+ *   Therefore [EvaluatedOrtIssue.pkg] is annotated with [JsonIdentityReference].
+ */
 data class EvaluatedModel(
-    val stats: Statistics
-    // TODO: add the actual evaluated data model.
+    val pathExcludes: List<PathExclude>,
+    val scopeExcludes: List<ScopeExclude>,
+    val copyrights: List<CopyrightStatement>,
+    val licenses: List<LicenseId>,
+    val scopes: List<ScopeName>,
+    val issueResolutions: List<IssueResolution>,
+    val issues: List<EvaluatedOrtIssue>,
+    val scanResults: List<EvaluatedScanResult>,
+    val packages: List<EvaluatedPackage>,
+    val paths: List<EvaluatedPackagePath>,
+    val dependencyTrees: List<DependencyTreeNode>,
+    val violationResolutions: List<RuleViolationResolution>,
+    val violations: List<EvaluatedRuleViolation>,
+    val declaredLicenseStats: SortedMap<String, Int>,
+    val detectedLicenseStats: SortedMap<String, Int>,
+    val statistics: Statistics,
+    val repositoryConfiguration: String,
+    val customData: CustomData
+) {
+    companion object {
+        private val INT_ID_TYPES = listOf(
+            CopyrightStatement::class.java,
+            EvaluatedOrtIssue::class.java,
+            EvaluatedPackage::class.java,
+            EvaluatedPackagePath::class.java,
+            EvaluatedRuleViolation::class.java,
+            EvaluatedScanResult::class.java,
+            IssueResolution::class.java,
+            LicenseId::class.java,
+            PathExclude::class.java,
+            RuleViolationResolution::class.java,
+            ScopeName::class.java,
+            ScopeExclude::class.java
+        )
+
+        private val MAPPER_CONFIG: ObjectMapper.() -> Unit = {
+            registerKotlinModule()
+
+            registerModule(JavaTimeModule())
+            registerModule(IntIdModule(INT_ID_TYPES))
+
+            disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+
+            propertyNamingStrategy = PROPERTY_NAMING_STRATEGY
+        }
+
+        private val JSON_MAPPER = JsonMapper().apply { MAPPER_CONFIG() }
+
+        fun create(input: ReporterInput): EvaluatedModel = EvaluatedModelBuilder(input).build()
+    }
+
+    fun toJson(): String {
+        val tree = JSON_MAPPER.valueToTree<ObjectNode>(this)
+        tree.forEach { node ->
+            if (node is ArrayNode) {
+                if (!node.isEmpty && node[0].has("_id")) {
+                    val sortedChildren =
+                        node.elements().asSequence().toList().sortedBy { it["_id"].intValue() }.toList()
```

Comment:
Nice! So this is the way to get rid of these viral `Sorted Collections`?!

Reply:
No, it's to ensure that the generated IDs match the array index, I will add a code comment.

Reply:
^^